yarn-project/p2p/src/services/reqresp/reqresp.ts

```diff
@@ -656,7 +660,9 @@ export class ReqResp {
         stream,
       );
     } finally {
-      await stream.close();
+      if (!streamAlreadyClosed) {
+        await stream.close();
+      }
```

Comment:
I'm wondering if we could use [`stream.status`](https://libp2p.github.io/js-libp2p/interfaces/_libp2p_interface.Stream.html#status) instead?